DomainContent/AvatarStore/CheckoutZone/checkoutZone.js

```diff
@@ -0,0 +1,255 @@
+
+
+        var newEntityProperties = Entities.getEntityProperties(newEntityID, ['marketplaceID', 'certificateID']);
+        var certificateID = undefined;
+        if (newEntityProperties.certificateID !== "" && newEntityProperties.certificateID !== undefined) {
+            certificateID = newEntityProperties.certificateID;
+        }
+        addTransformForMarketplaceItem(newEntityProperties.marketplaceID, certificateID, transform);
+
+        // Remove the demo object, to prevent overlapping objects
+        Entities.deleteEntity(transform.demoEntityID);
+    };
+
+    _this.enterEntity = (function (entityID) {
+        isInZone = true;
+        left = true;
+        getCheckoutStandPosition();
+        SHARED.getAvatarChildEntities(MyAvatar).forEach(function (entityID) {
+            var maxItems = 10;
+            if (replicaList.length < maxItems){
+                var childUserData = Entities.getEntityProperties(entityID, 'userData').userData;
+                var isAttachment = childUserData.indexOf("attached\":true");
+                var marketplaceID = Entities.getEntityProperties(entityID, 'marketplaceID').marketplaceID;
+                if (marketplaceID && (isAttachment !== -1)) {
+                
+                    // TODO check for already purchased 
+                    spawnOverlayReplica(entityID); // put a copy of the item on the table
+                    // move spawn position over to the next empty spot
+                    var moveRight = 1.5 * ITEM_HEIGHT;
+                    var moveBack = 0.75 * ITEM_HEIGHT;
+                    var moveDown = 1.25 * ITEM_HEIGHT;
+                    var moveLeft = 1.5 * ITEM_HEIGHT;
+                    var moveForward = 0.75 * ITEM_HEIGHT;
+                    if (left) {
+                        spawnZ += moveRight;
+                        spawnX += moveBack;
+                        left = false;
+                    } else {
+                        spawnY -= moveDown;
+                        spawnZ -= moveLeft;
+                        spawnX -= moveForward;
+                        left = true;
+                    }
+                }
+            }
+
+            Overlays.mousePressOnOverlay.connect(function(overlayID, event){
+                if (replicaList.indexOf(overlayID) !== -1) {
+                    var name = Overlays.getProperty(overlayID, 'name');
+                    var id = name.substr(OVERLAY_PREFIX.length, OVERLAY_PREFIX.length + 35);
+                    var goToURL = MARKET_PLACE_ITEM_URL_PREFIX + id;
+                    var tablet = Tablet.getTablet("com.highfidelity.interface.tablet.system");
+                    tablet.gotoWebScreen(goToURL);
+                }
+            });
+        });
+    });
+    
+    _this.leaveEntity = function() {
+        isInZone = false;
+        Entities.findEntities(MyAvatar.position, 1000).forEach(function(entity) {
```

Comment:
We may want to reconsider this behavior. Right now, if you purchase something you're wearing but don't rez it (not ideal case, but one we should handle) leave the zone, and then rez it, we don't rez it in the right place. We can't protect this entirely but we could have a zone for the entire domain handle the rez after purchase functionality, in which case we'd want these transforms to remain to reference at a later time.

Reply:
We will come up with something before thanksgiving, not MVP for now.